src/App.tsx

```diff
@@ -1,90 +1,175 @@
-import { Contract } from "web3-eth-contract";
-import { Stream } from "./typings/types";
+import { SafeInfo, SdkInstance } from "@gnosis.pm/safe-apps-sdk";
+import { BigNumber, Contract, ethers, utils } from "ethers";
 
-import { SelectContainer, ButtonContainer } from "./components";
 import StreamLengthInput, { StreamLength } from "./components/StreamLengthInput";
 import WidgetWrapper from "./components/WidgetWrapper";
-
-import ERC20Abi from "./abis/erc20";
-import { web3Provider, getTokenList, TokenItem } from "./config/config";
+import erc20Abi from "./abis/erc20";
 import createStreamTxs from "./transactions/createStream";
-import getStreams from "./utils/streams";
+import getStreams from "./gql/streams";
+import provider from "./config/provider";
 import theme from "./theme";
 
-const web3: any = new Web3(web3Provider);
+import { SelectContainer, ButtonContainer } from "./components";
+import { Stream, TransactionList } from "./typings/types";
+import { getTokenList, TokenItem } from "./config/tokens";
+import { useAppsSdk } from "./hooks";
 
 const StyledTitle = styled(Title)`
   margin-top: 0px;
 `;
 
-const SablierWidget = () => {
-  const [safeInfo, setSafeInfo] = useState<SafeInfo>();
-  const [tokenList, setTokenList] = useState<Array<TokenItem>>();
-
+function SablierWidget() {
+  /*** State Variables ***/
+  const [appsSdk, safeInfo]: [SdkInstance, SafeInfo | undefined] = useAppsSdk();
+  const [amountError, setAmountError] = useState<string | undefined>();
+  const [outgoingStreams, setOutgoingStreams] = useState<Stream[]>([]);
   const [recipient, setRecipient] = useState<string>("");
-
   const [selectedToken, setSelectedToken] = useState<TokenItem>();
+  const [streamAmount, setStreamAmount] = useState<string>("");
+  const [streamLength, setStreamLength] = useState<StreamLength>({ days: "0", hours: "0", minutes: "0" });
+  const [streamLengthError, setStreamLengthError] = useState<string | undefined>();
+  const [tokenBalance, setTokenBalance] = useState<string>("0");
   const [tokenInstance, setTokenInstance] = useState<Contract>();
+  const [tokenList, setTokenList] = useState<TokenItem[]>();
 
-  const [tokenBalance, setTokenBalance] = useState<string>("0");
+  /*** Callbacks ***/
 
-  const [streamLength, setStreamLength] = useState<StreamLength>({ days: "0", hours: "0", minutes: "0" });
-  const [streamLengthError, setStreamLengthError] = useState<string | undefined>();
+  const bigNumberToHumanFormat = useCallback(
+    (value: string): string => {
+      if (!selectedToken) {
+        return "";
+      }
+      let scaledNumber = utils.formatUnits(value, selectedToken.decimals);
+
+      // Pad with zeros for 4 decimal places
+      // Note: `formatUnits` always returns a decimal point
+      scaledNumber = scaledNumber.padEnd(scaledNumber.indexOf(".") + 5, "0");
+      // Trim any excess decimal places
+      return scaledNumber.slice(0, scaledNumber.indexOf(".") + 5);
```

Comment:
Here's the less-than-pretty string manipulation. We can cut the padding if we are happy to show fewer decimal places in certain situations.